### PR TITLE
Fix crash when `isort` isn't installed

### LINKED ISSUE #130: Crashes if isort isn't installed
conda-forge/darker-feedstock#1 fails with this error (in the Linux and OSX builds):
```
+ darker --help
Traceback (most recent call last):
  File "/home/conda/feedstock_root/build_artifacts/darker_1619998672207/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/bin/darker", line 7, in <module>
    from darker.__main__ import main
  File "/home/conda/feedstock_root/build_artifacts/darker_1619998672207/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.9/site-packages/darker/__main__.py", line 16, in <module>
    from darker.import_sorting import apply_isort, isort
  File "/home/conda/feedstock_root/build_artifacts/darker_1619998672207/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.9/site-packages/darker/import_sorting.py", line 22, in <module>
    isort_code = getattr(isort, "code")
AttributeError: 'NoneType' object has no attribute 'code'
```

### PR DESCRIPTION
**Pull request by akaihola**
Fixes #130.

### CODE REVIEW

**Comment by akaihola** [comment]
@martinRenou, would you like to review this fix? I had to refactor the tests a bit, so if it looks too heavy to review, I can ask some other contributors instead.

**Comment by martinRenou** [comment]
I can try to find time, but I am no expert in black/darker :S

File: src/darker/import_sorting.py [hunk 1]
```diff
@@ -15,11 +15,21 @@
 
 try:
     import isort
+
+    # Work around Mypy problem
+    # https://github.com/python/mypy/issues/7030#issuecomment-504128883
+    isort_code = getattr(isort, "code")
```
Comment (samoylovfp) [inline src/darker/import_sorting.py:21]: getattr throws an `AttributeError` when attribute is not available. You probably want to either catch that as well or add a default argument to the `getattr` call

File: src/darker/tests/conftest.py [hunk 2]
```diff
@@ -14,15 +15,23 @@
...
+def isort_present():
+    """Fixture for removing or adding the `isort` package temporarily for a test"""
+
+    @contextmanager
+    def _isort_present(present):
+        if present:
+            # Inject a dummy `isort` package temporarily
+            fake_isort_module: Optional[types.ModuleType] = types.ModuleType("isort")
+            # dummy function required by `import_sorting`:
+            fake_isort_module.code = None  # type: ignore
```
Comment (samoylovfp) [inline src/darker/tests/conftest.py:27]:
```suggestion
            del fake_isort_module.code
```

File: src/darker/import_sorting.py (hunk 1 above)
Comment (samoylovfp) [reply src/darker/import_sorting.py:21]: image

Comment (samoylovfp) [reply src/darker/import_sorting.py:21]: And I just realized you were fixing a different problem

**Comment by akaihola** [comment]
@samoylovfp do you approve these changes? I noticed you opened and closed two conversations.

File: src/darker/tests/conftest.py [hunk 3]
```diff
@@ -14,15 +15,23 @@
...
-def without_isort():
-    with patch.dict(sys.modules, {"isort": None}):
-        yield
-
-
-@pytest.fixture
-def with_isort():
-    with patch.dict(sys.modules, {"isort": types.ModuleType("isort")}):
-        yield
+def isort_present():
```
Comment (samoylovfp) [inline src/darker/tests/conftest.py:18]: Not sure if it makes sense to wrap this into a fixture, it could just be the context manager

**Review by samoylovfp — APPROVED** [review]

**Comment by samoylovfp** [comment]
@akaihola I started reviewing, found issues, then realized that I misunderstood the change and decided to take a break.

Now looked through it with a fresh eye and it all looks good modulo a nitpick

File: src/darker/tests/conftest.py (hunk 3 above)
Comment (akaihola) [reply src/darker/tests/conftest.py]: Absolutely. Thanks!

**Pull request merged**